feat(Task-8): As a User, I want a good MainPage

jLagosCarrera opened the pull request.

jLagosCarrera committed:
feat(Task-1): Configure Webpack to our workflow

jLagosCarrera committed:
feat(Task-1): Add AngularJS hello world

jLagosCarrera committed:
feat(Task-1): Split config files for development and production

jLagosCarrera committed:
feat(Task-1): Add HtmlWebpackPlugin to serve index file

jLagosCarrera committed:
feat(Task-1): Add dist folder to gitignore

jLagosCarrera committed:
feat(Task-4): Create service for handling OWM request and use it on the component

jLagosCarrera committed:
Merge branch 'development' into feat/Task-4

jLagosCarrera committed:
Merge branch 'development' into feat/Task-4

jLagosCarrera committed:
refactor(Task-4): Structure all code to be more mantainable

jLagosCarrera committed:
feat(Task-4): Use of plugin to hide sensible information

jLagosCarrera committed:
feat(Task-5): Cleared all code from previous config tasks

jLagosCarrera committed:
Refactor(Task-5): Clarified a few lines

jLagosCarrera committed:
feat(Task-7): Start home page screen look and feel

jLagosCarrera committed:
feat(Task-7): Add input field and search button with validation

jLagosCarrera committed:
feat(Task-7): Created list for recent searches and managed save and load it on localStorage

jLagosCarrera committed:
refactor(Task-7): Restructured and adapted all code to ES6

jLagosCarrera committed:
feat(Task-7): Make API calls

jLagosCarrera committed:
Merge branch 'development' into feat/Task-7

jLagosCarrera committed:
feat(Task-8): Done routing

jLagosCarrera committed:
feat(Task-8): Fix minor bug on routing and started page look and feel

jLagosCarrera committed:
feat(Task-8): Add link to homepage and search form

jLagosCarrera committed:
refactor(Task-8): Made input and button in same line on search page

jLagosCarrera committed:
feat(Task-8): Display searched city

jLagosCarrera committed:
Merge branch 'development' into feat/Task-8

jLagosCarrera committed:
feat(Task-8): Finished screen dividing and look and feel

eloydotco commented on src/app/home-page/home-page.controller.js:
```diff
@@ -10,14 +11,17 @@ export default class HomePageController {
-        //Will do this call on the search page
-        //this.openWeatherMaps.getFiveDayForecastCity(this.city);
+        this.openWeatherMaps.getCurrentForecastCity(this.city);
+        this.openWeatherMaps.getFiveDayForecastCity(this.city);
```
This logic should be called on the init method of the search component, because if the user refresh the page (or access directly) on /search, the current and future forecast won't be available.
If you have doubts don't hesitate to contact me :)

jLagosCarrera replied:
Sure, I undertand, messed up everything a bit, changing it right now

eloydotco commented on src/app/home-page/home-page.controller.js:
```diff
@@ -10,14 +11,17 @@ export default class HomePageController {
     clickCity(city) {
-        //Will do this call on the search page
-        //this.openWeatherMaps.getFiveDayForecastCity(city);
+        this.latestSearchesUtil.addCity(city);
```
this method does exactly the same as `search()`, avoid write redundant lines of code, you can delete this method, or just call the other

jLagosCarrera replied:
Noticed on next task, Ill solve it here, havent done it yet on next one

diegopf commented on src/app/app.module.js:
```diff
@@ -1,14 +1,15 @@
+import angularmessages from 'angular-messages';
 import 'angular-material/angular-material.scss';
-import 'material-icons/iconfont/material-icons.scss';
+import 'material-icons/iconfont/material-icons.scss'
```
missing semicolon

eloydotco commented on src/app/shared/services/latestSearchesUtil.js:
```diff
@@ -7,11 +7,20 @@ export default class LatestSearchesUtil {
-    addCity(city) {
-        if (!city || city.trim() === '' || this.latestSearches.includes(city)) {
+    addCity(rawCity) {
+        let city = rawCity.charAt(0).toUpperCase() + rawCity.slice(1).toLowerCase();
```
Check if rawCity is not defined (if the user clicks on search button with empty input, an error appears on the console)

diegopf replied:
also, why are you doing that at all?

jLagosCarrera replied:
Woah, this one was strange, I already solved, didn't know why I did that, sorry!

eloydotco commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -19,7 +19,7 @@ export default class OpenWeatherMaps {
 
         try {
-            return this.fiveDayForecast = await this.$http.get(`${this.baseURL}/weather`, params);
+            return this.currentForecast = await this.$http.get(`${this.baseURL}/weather`, params);
```
it's better split this line into two -> assign statement and the return statement
```
this.currentForecast = await this.$http.get(`${this.baseURL}/weather`, params);
return this.currentForecast;

jLagosCarrera replied:
May I ask why? Already changed anyway

eloydotco replied:
Just better readability for other developers :)
on the first look I didn't realized up of return statement, just the variable assigment

eloydotco commented on src/app/shared/services/latestSearchesUtil.js:
```diff
@@ -7,11 +7,20 @@ export default class LatestSearchesUtil {
         }
 
+        if (this.latestSearches.includes(city)) {
+            let index = this.latestSearches.indexOf(city);
```
index can be a const

eloydotco commented on src/app/app.module.js:
```diff
@@ -1,14 +1,15 @@
+import angularmd from 'angular-material';
+import angularanimate from 'angular-animate';
+import angulararia from 'angular-aria';
+import angularmessages from 'angular-messages';
```
These names were right, camelCase format is better

jLagosCarrera replied:
Maybe did a bad merge from dev, sorry!

jLagosCarrera committed:
fix(Task-8): Solved requested changes

Camelcase names, duplicated code, nonsense variable declaration and split await return

jLagosCarrera commented:
All requested changes were made

jLagosCarrera committed:
fix(Task-8): Change code capitalize for css capitalize

gbtec-ext-jorgevillar commented on src/app/search-page/search-page.scss:
```diff
@@ -0,0 +1,21 @@
+    height: 100%;
+    background-image: url("../../../img/contentBackground.svg");
+    background-position: center;
+    background-repeat: no-repeat;
```
You can create a mixin for this to avoid repeatin this 4 styles
